ui/app/conf-tx.js

```diff
@@ -172,47 +172,53 @@ ConfirmTxScreen.prototype.onTxChange = function (txData) {
 // Must default to any local state txData,
 // to allow manual override of gas calculations.
 ConfirmTxScreen.prototype.sendTransaction = function (txData, event) {
-  event.stopPropagation()
+  this.stopPropagation(event)
   const state = this.state || {}
   const txMeta = state.txData
   this.props.dispatch(actions.updateAndApproveTx(txMeta || txData))
 }
 
 ConfirmTxScreen.prototype.cancelTransaction = function (txData, event) {
-  event.stopPropagation()
+  this.stopPropagation(event)
   this.props.dispatch(actions.cancelTx(txData))
 }
 
 ConfirmTxScreen.prototype.signMessage = function (msgData, event) {
   log.info('conf-tx.js: signing message')
   var params = msgData.msgParams
   params.metamaskId = msgData.id
-  event.stopPropagation()
+  this.stopPropagation(event)
   this.props.dispatch(actions.signMsg(params))
 }
 
+ConfirmTxScreen.prototype.stopPropagation = function (event) {
+  if (event.stopPropagation) {
```

Comment:
Correct.  Specifically, the `invalid` event emitted by forms, so I just made this function so we could handle it generically with our other events in this component.